chainerx_cc/chainerx/backward.cc

```diff
@@ -211,27 +213,34 @@ class BackwardImpl {
           backprop_id_{backprop_id},
           double_backprop_{double_backprop},
           array_node_grad_map_{std::move(array_node_grad_map)} {
+        // The use of non-POD static storage object here is safe, because destructing a shared_ptr with nullptr does not incur any
+        // destruction order problem.
+        static const std::shared_ptr<ArrayNode> kNullArrayNode{nullptr};
+
         for (const Array& output : outputs) {
-            if (!output.IsBackpropRequired(backprop_id)) {
-                throw ChainerxError{"Cannot start backprop from an array whose gradient is not required (on graph '", backprop_id, "')"};
+            if (output.IsBackpropRequired(backprop_id)) {
+                output_array_nodes_.emplace_back(internal::GetArrayBody(output)->GetArrayNode(backprop_id));
+            } else {
+                output_array_nodes_.emplace_back(kNullArrayNode);
             }
-            output_array_nodes_.emplace_back(internal::GetArrayBody(output)->GetArrayNode(backprop_id));
         }
 
-        // Check if backward is possible for the given graph, in this context.
-        // It is not possible if a graph from an outer scope has already been backpropped.
-        backprop_id.context().CheckBackpropAllowed(backprop_id);
+        if (!output_array_nodes_.empty()) {
```

Comment:
A minor but `output_array_nodes_` should have the same size here as `outputs`. If `outputs` is empty, this ctor body does essentially nothing. How about checking that in the beginning with an early return?

Reply:
Done.